### ISSUE TITLE
Why not MVI?

### ISSUE THREAD

**Issue by StefanOltmann** [Kotlin/kmm-production-sample#22]
Just out of curiosity: Why did you choose the minimalistic redux instead of MVI?

I know this is all pretty similar and mostly names differ, but the jb-compose samples use MVI.

**Comment by KaterinaPetrova** [comment]
Hey @StefanOltmann ! It's the beauty of KMM — the freedom of choice 😃 At the heart of it all is just our lovely pure Kotlin (with some multiplatform magic 🧙🏻‍♂️), so we don't have any restrictions either on how we create the UI layer, either on what architecture we use on the lower layers (and which of them do we want to share). 
@terrakok and I chose this minimalistic redux approach because we enjoyed using it while we were developing user-facing mobile apps. From our point of view, it works great for clientside apps. On the one hand, it allows to minimize the number of errors related to invalid state UI (especially when the user interacts with the UI), and on the other hand it's quite simple and doesn't require to spend much time for a new developer to start using it (plus no additional dependencies!)
